src/mrinufft/trajectories/tools.py

```diff
@@ -555,24 +613,43 @@ def get_gradient_amplitudes_to_travel_for_set_time(
             gmax=gmax,
             smax=smax,
         )
-        N = (
-            np.max(np.sum(n_ramp_down + n_ramp_up + n_plateau, axis=0)) + 2
-        )  # Extra 2 buffer samples
+        N = np.max(n_ramp_down + n_ramp_up + n_plateau) + 2
 
     area_needed = (ke - ks) / gamma / raster_time
     # Intermediate gradient values. This is value of plateau or triangle gradients
     gi = np.zeros_like(ks, dtype=np.float32)
 
+    # Assume direct solution first
+    n_ramp_up = np.ones(gs.shape, dtype=int) * N // 2
+    n_ramp_down = N - n_ramp_up
+    gi = (2 * area_needed - (n_ramp_down + 1) * gs - (n_ramp_up - 1) * ge) / (
+        n_ramp_down + n_ramp_up
+    )
+    max_slew_needed = raster_time * np.max(
+        [abs(gi - gs) / n_ramp_down, abs(ge - gi) / n_ramp_up], axis=0
+    )
+    # FIXME: Becareful of rotating FOV boxes.
```

Comment:
This is a general bug in all the pre-phasors codes. These pre-phasors assume Smax indivitually in each direction. However, that results in limited trajectories which cant really rotate! 
As we dont have FOV boxes, the only way this can really be handeled is in sequence itself (or a very constrained pre-phasor / rewinders, which Smax reduced to Smax / np.sqrt(3). 
Ill have a chat with Franck regarding this and see if it can be updated. This should be tracked in different issue.